src/auth/session/storage/custom.ts

```diff
@@ -35,6 +35,10 @@ export class CustomSessionStorage implements SessionStorage {
     if (result) {
       if (result instanceof Session) {
         return result;
+      } else if (result instanceof Object && 'id' in result) {
+        let session = new Session(result.id as string);
+        session = Object.assign(session, result);
```

Comment:
I think the more 'modern' way of doing this would be (untested and unsure if the result will actually be a `Session`):
```suggestion
        session = {...session, ...result};
```

Reply:
I'll try it out, but I'm not 100% sure this will work.

Reply:
TIL this totally does work this way.

Reply:
`session = {...session, ...result};`

I think the more type-strict way of doing this.
`session = Session.cloneSession(result, result.id);`